### PR TITLE
Switch to using config/secrets.toml.

### PR DESCRIPTION
**Pull request by brianwgoldman**
Also included in this change:

* If the config/ directory doesn't exist, create stub files in it.
* Loading secrets happens before all commands, simplifying command setup.
* Tool for migrating secrets/default.json to the new toml file.

### CODE REVIEW

**Review by dhosterman — APPROVED** [review]
I think this looks good! I do like the idea of being able to define a location for the secrets other than relative to the current directory, so I'm a little sad to see that CLI option go, but I think it'll be easy enough to add later when we're more confident in the direction we'll ultimately take with it.

**Review by wpietri — APPROVED** [review]
Looks great! I'm all for it.

File: newhelm/config_templates/secrets.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Edit this file to add your secrets.
+
+# This is an example of how to define a secret.
+# The config is saying that within scope "demo" we have a
+# key named "demo_api_key" that we are setting to value "12345".
+[demo]
```
[the remark this answers is not in the document]
Comment (brianwgoldman) [reply newhelm/config_templates/secrets.toml:6]: If we make them live, we switch errors from "missing the key" to "this key is invalid". My preference is to keep the former. I've added commented out structures for perspective and together.

File: tests/test_config.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def test_write_default_config_writes_files(tmpdir):
+    config_dir = tmpdir.join("config")
+    write_default_config(config_dir)
+    files = [f.basename for f in config_dir.listdir()]
+    assert files == ["secrets.toml"]
+
+
+def test_write_default_config_skips_existing_dir(tmpdir):
```
[the remark this answers is not in the document]
Comment (brianwgoldman) [reply tests/test_config.py:17]: My reasoning was that if someone was going through the trouble of setting it up, we wouldn't want to pollute their progress with unexpected files. This is also protection against clashing with an existing `config` directory that someone created for some other purpose.

Given that we don't have immediate plans to add any other config files, I think the difference here is small enough to not worry about.

**Pull request merged**